src/pages/workspaces/workspace/tools/WorkflowView.js

```diff
@@ -500,27 +507,30 @@ const WorkflowView = _.flow(
     Utils.withBusyState(v => this.setState({ updatingConfig: v }))
   )(async newSnapshotId => {
     const { ajax: { Methods } } = this.props
-    const { modifiedConfig: { methodRepoMethod: { methodNamespace, methodName } } } = this.state
+    const { modifiedConfig: { methodRepoMethod: { methodNamespace, methodName } }, currentSnapRedacted } = this.state
     const config = await Methods.template({ methodNamespace, methodName, methodVersion: newSnapshotId })
     const modifiedInputsOutputs = await Methods.configInputsOutputs(config)
-    this.setState({ modifiedInputsOutputs: this.sortOptionalInputs(modifiedInputsOutputs) })
+    this.setState(
+      { modifiedInputsOutputs: this.sortOptionalInputs(modifiedInputsOutputs), savedSnapRedacted: currentSnapRedacted, currentSnapRedacted: false })
     this.setState(_.set(['modifiedConfig', 'methodRepoMethod'], config.methodRepoMethod))
+    this.fetchInfo(config)
```

Comment:
FWIW I suspect there's an existing issue with the WDL not getting refreshed if you cancel out of a version change. We could fix that as part of this, or separately.

Reply:
Gonna do it here, since I discovered it working on this so it's relevant.